client/src/main/java/org/eclipse/hono/client/CommandClient.java

```diff
@@ -0,0 +1,54 @@
+package org.eclipse.hono.client;
+
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * A client for accessing Hono's Command and Control API.
+ * <p>
+ * An instance of this interface is always scoped to a specific tenant.
+ * </p>
+ */
+public interface CommandClient extends RequestResponseClient {
+
+    /**
+     * Sends a command to a device and expects a response.
+     * <p>
+     * A device needs to be (successfully) registered before a client can upload
+     * any data for it. The device also needs to be connected for a successful delivery.
+     *
+     * @param command The command name.
+     * @param data The command data to send to the device.
+     * @return A future indicating the result of the operation.
+     *         <p>
+     *         The future will succeed if a response with status 201 has been received from the device/adapter.
+     *         <p>
+     *         Otherwise, the future will fail with a {@link ServiceInvocationException} containing
+     *         the (error) status code returned by the service.
+     * @throws NullPointerException if device ID is {@code null}.
+     * @see RequestResponseClient#setRequestTimeout(long)
+     */
+    Future<Buffer> commandWithResponse(String command, Buffer data);
+
+
+    /**
+     * Sends a command to a device and expects a response.
+     * <p>
+     * A device needs to be (successfully) registered before a client can upload
+     * any data for it. The device also needs to be connected for a successful delivery.
+     *
+     * @param command The command name.
+     * @param data The command data to send to the device.
+     * @return A future indicating the result of the operation.
+     *         <p>
+     *         The future will succeed if a response with status 201 has been received from the device/adapter.
+     *         <p>
+     *         Otherwise, the future will fail with a {@link ServiceInvocationException} containing
+     *         the (error) status code returned by the service.
+     * @throws NullPointerException if device ID is {@code null}.
```

Comment:
there is no device ID parameter, which is very strange because it raises the question: how do we address a device?

I am a little confused now, is this client supposed to be used by an application that wants to send a command to a device?

Reply:
Yes, but the link is created exclusively for this device - so there is no need to set it again at sending.

Reply:
I see, then you should at least update the interface's class level JavaDoc accordingly. It states that the client is scoped to a tenant.